### PR TITLE
feat(12-factor-app): add configurations to deploy as 12 factor app [WD-11371]

### PR DESCRIPTION
**Pull request by lorumic**
## Done

- Added configuration files (`rockcraft.yaml`, `charmcraft.yaml`) and `deploy.yaml` workflow to deploy the website as a 12-factor application.

## Issue / Card

Fixes [WD-11371](https://warthogs.atlassian.net/browse/WD-11371)

[WD-11371]: https://warthogs.atlassian.net/browse/WD-11371?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        run: |
+          yarn install
+          yarn run build-css
+          yarn run build-js
+
+      - name: Setup LXD
+        uses: canonical/setup-lxd@main
+
+      - name: Setup Rockcraft
+        run: sudo snap install rockcraft --classic --channel=latest/edge
```
Comment (edlerd) [inline .github/workflows/deploy.yaml:53]: Should we use a stable channel here?

**Review by edlerd — COMMENTED** [review]
Can we reference the `charm/lib/charms` directory as a library in a remote repository? Otherwise, we end up copying into each website and maintaining becomes very hard.

**Comment by lorumic** [comment]
> Can we reference the `charm/lib/charms` directory as a library in a remote repository? Otherwise, we end up copying into each website and maintaining becomes very hard.

I think we should probably exclude the `lib` folder and its content from version control. These are libraries that are fetched when running the `charmcraft fetch-libs` command, so we can consider them as "build artifacts" in a sense.

Regarding the remote repository idea, I don't think that can work, because each website can have different dependencies.

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (lorumic) [reply .github/workflows/deploy.yaml:53]: I'd say the outcome of the discussion is that, for the time being, keeping using the `edge` channel is the safest choice.

File: konf/site.yaml [hunk 2]
```diff
@@ -6,11 +6,16 @@ env:
   - name: SENTRY_DSN
     value: https://37706e022f2841448dbb094990420522@sentry.is.canonical.com//23
 
-  - name: SEARCH_API_KEY
+  - name: FLASK_GOOGLE_SEARCH_API_KEY
     secretKeyRef:
       key: google-custom-search-key
       name: google-api
 
+  - name: FLASK_SECRET_KEY
```
Comment (edlerd) [inline konf/site.yaml:14]:
```suggestion
  - name: SECRET_KEY
```
Why don't we call it `SECRET_KEY` here? Then we don't need to alias it in app.py

Comment (lorumic) [reply konf/site.yaml:14]: [Here](https://docs.google.com/document/d/1zOmyP6FyMQMhGNOs69O7bVGpPSi110z-KQ8cG_Phe1s/edit#heading=h.ehuw67mkj6pz) is the answer to your question.

**Review by edlerd — APPROVED** [review]
LGTM :+1:

**Pull request merged**